projects/dotcms-js/src/lib/core/core-web.service.ts

```diff
@@ -48,40 +50,53 @@ export const RULE_CONDITION_UPDATE_TYPE = 'RULE_CONDITION_UPDATE_TYPE';
 export const RULE_CONDITION_UPDATE_PARAMETER = 'RULE_CONDITION_UPDATE_PARAMETER';
 export const RULE_CONDITION_UPDATE_OPERATOR = 'RULE_CONDITION_UPDATE_OPERATOR';
 
+export interface RequestOptionsParams {
+    headers?: HttpHeaders;
+    reportProgress?: boolean;
+    params?: HttpParams;
+    responseType?: 'arraybuffer' | 'blob' | 'text' | 'json';
+    withCredentials?: boolean;
+}
+
 @Injectable()
 export class CoreWebService {
     private httpErrosSubjects: Subject<any>[] = [];
 
     constructor(
         private _apiRoot: ApiRoot,
-        private _http: Http,
         private loggerService: LoggerService,
         private browserUtil: BrowserUtil,
-        private router: Router
+        private router: Router,
+        private http: HttpClient
     ) {}
 
-    request(options: any): Observable<any> {
+    request(options: RequestOptionsArgs): Observable<any> {
```

Comment:
```suggestion
    request<T>(options: DotRequestOptionsArgs): Observable<T | any> {
```

I think is time to do this, not sure if this works but we can take the httpclient approach with this.